app/src/main/java/nl/tudelft/cs4160/trustchain_android/main/PeerListAdapter.java

```diff
@@ -179,30 +185,27 @@ private String getLastSeen(long msSinceLastMessage) {
      */
     private void setOnClickListener(TableLayout mTableLayoutConnection, int position) {
         mTableLayoutConnection.setTag(position);
-        View.OnClickListener onClickListener = new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                int pos = (int) v.getTag();
-                Peer peer = getItem(pos);
-                if(peer.isAlive() && peer.hasReceivedData()) {
-                    PublicKeyPair pubKeyPair = PubKeyAndAddressPairStorage.getPubKeyByAddress(context, peer.getAddress().toString().replace("/", ""));
-                    if(pubKeyPair != null) {
-                        InboxItem i = new InboxItem(peer.getPeerId(), new ArrayList<Integer>(), peer.getAddress().getHostString(), pubKeyPair, peer.getPort());
-                        UserNameStorage.setNewPeerByPublicKey(context, peer.getPeerId(), pubKeyPair);
-                        InboxItemStorage.addInboxItem(context, i);
-                        Snackbar mySnackbar = Snackbar.make(coordinatorLayout,
-                                peer.getPeerId() + " added to inbox", Snackbar.LENGTH_SHORT);
-                        mySnackbar.show();
-                    } else {
-                        Snackbar mySnackbar = Snackbar.make(coordinatorLayout,
-                                "This peer didn't send a public key yet", Snackbar.LENGTH_SHORT);
-                        mySnackbar.show();
-                    }
+        View.OnClickListener onClickListener = v -> {
+            int pos = (int) v.getTag();
+            Peer peer = getItem(pos);
+            if(peer.isAlive() && peer.isReceivedFrom()) {
+                PublicKeyPair pubKeyPair = PubKeyAndAddressPairStorage.getPubKeyByAddress(context, peer.getAddress().toString().replace("/", ""));
+                if(pubKeyPair != null) {
+                    InboxItem i = new InboxItem(peer, new ArrayList<>());
+                    UserNameStorage.setNewPeerByPublicKey(context, peer.getName(), pubKeyPair);
+                    InboxItemStorage.addInboxItem(context, i);
+                    Snackbar mySnackbar = Snackbar.make(coordinatorLayout,
```

Comment:
Should we add a function that takes as parameter the string to a snackbar and shows the snackbar on execution?

Reply:
Nah, I don't think so, this is only one line at the time. I did move the strings to the strings.xml though.